### PR TITLE
fix(workflows): Properly use caches in AWS, Azure, GCP, K8s workflows

### LINKED ISSUE #9388: bug(aws,gcp,k8s,azure): Test caching in CI does not work
### Which problem is this feature request solving?

Every time we update huge plugins (like AWS) we retest all pf the code instead of only testing the affected parts.

### Describe the solution you'd like

Only testing the affected parts (need to make sure, though, that the libraries updates trigger the tests, too).
Possibly, this can be achieved by caching Go build cache & test cache (see [this](https://github.com/mvdan/github-actions-golang/blob/master/.github/workflows/test.yml) workflow as an example).

Relevant info:
https://pkg.go.dev/cmd/go#hdr-Build_and_test_caching
https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows
https://github.com/marketplace/actions/cache

https://github.com/marketplace/actions/golang-with-cache
https://github.com/mvdan/github-actions-golang

### Pull request (optional)

- [x] I can submit a pull request

### PR DESCRIPTION
**Pull request by candiduslynx**
Short way instead of #9656.
Closes #9388

### CODE REVIEW

File: .github/workflows/source_aws.yml [hunk 1]
```diff
@@ -135,11 +136,12 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v3
       - name: Set up Go 1.x
-        uses: actions/setup-go@v3
+        uses: erezrokah/setup-go@feat/add_cache_prefix
         with:
           go-version-file: plugins/source/aws/go.mod
           cache: true
           cache-dependency-path: plugins/source/aws/go.sum
+          cache-prefix: policies-cache-
```
Comment (erezrokah) [inline .github/workflows/source_aws.yml:144]:
```suggestion
          cache-key-prefix: policies-cache-
```

File: .github/workflows/source_azure.yml [hunk 2]
```diff
@@ -117,11 +118,12 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v3
       - name: Set up Go 1.x
-        uses: actions/setup-go@v3
+        uses: erezrokah/setup-go@feat/add_cache_prefix
         with:
           go-version-file: plugins/source/aws/go.mod
           cache: true
           cache-dependency-path: plugins/source/aws/go.sum
+          cache-prefix: policies-cache-
```
Comment (erezrokah) [inline .github/workflows/source_azure.yml:126]:
```suggestion
          cache-key-prefix: policies-cache-
```

File: .github/workflows/source_k8s.yml [hunk 3]
```diff
@@ -117,11 +118,12 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v3
       - name: Set up Go 1.x
-        uses: actions/setup-go@v3
+        uses: erezrokah/setup-go@feat/add_cache_prefix
         with:
           go-version-file: plugins/source/k8s/go.mod
           cache: true
           cache-dependency-path: plugins/source/k8s/go.sum
+          cache-prefix: policies-cache-
```
Comment (erezrokah) [inline .github/workflows/source_k8s.yml:126]:
```suggestion
          cache-key-prefix: policies-cache-
```

**Review by erezrokah — CHANGES_REQUESTED** [review]
I thinks something is not working with my cache key fix, see https://github.com/cloudquery/cloudquery/actions/runs/4659445116/jobs/8246354513?pr=9812#step:14:7

image

Maybe try directly with the cache action instead of my fork?

**Comment by candiduslynx** [comment]
> I thinks something is not working with my cache key fix, see https://github.com/cloudquery/cloudquery/actions/runs/4659445116/jobs/8246354513?pr=9812#step:14:7
> 
> image
> 
> Maybe try directly with the cache action instead of my fork?

I think it'll be fixed by d685b03

**Comment by erezrokah** [comment]
It happens on GCP too:
https://github.com/cloudquery/cloudquery/actions/runs/4659445149/jobs/8246354731#step:14:7
https://github.com/cloudquery/cloudquery/actions/runs/4659445149/jobs/8246353443#step:17:7
image

**Comment by candiduslynx** [comment]
@erezrokah it's fine now?

**Comment by erezrokah** [comment]
> @erezrokah it's [fine](https://github.com/cloudquery/cloudquery/actions/runs/4659711753/jobs/8246902366?pr=9812) now?

This shows a cache hit, so not the same test as it skips saving the cache. Can we find the workflow that saved that cache successfully to confirm? Or delete the cache manually and confirm each job saves it successfully? Another option to test is to add `v1` or something to the prefix to create a new key.

**Comment by candiduslynx** [comment]
I'm removing the caches designated via [this](https://github.com/cloudquery/cloudquery/actions/caches?query=branch%3Arefs%2Fpull%2F9812%2Fmerge) branch and rerunning

**Comment by candiduslynx** [comment]
Now triggering build

**Comment by erezrokah** [comment]
Also https://github.com/cloudquery/cloudquery/pull/9812/commits/d4f2f23542a59284b155ef995f43c40d25dc8168. If it doesn't pass we can descope it from this PR

**Review by erezrokah — APPROVED** [review]

**Pull request merged**